### PR TITLE
Feat/2582 savesets in dashboard

### PR DESCRIPTION
**Pull request by adipaul1981**

### CODE REVIEW

File: src/components/UserDashboard/ParticipantSets/ParticipantSets.css [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.ant-table-tbody > tr > td {
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/ParticipantSets.css:1]: should it be in the global style?

File: src/components/UserDashboard/ParticipantSets/Test/ParticipantSets.test.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import ParticipantSets from '../index';
+import { configure, mount, ReactWrapper } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { LoggedInUser } from '../../../../store/userTypes';
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/Test/ParticipantSets.test.tsx:5]: fix import path ../../

File: src/components/UserDashboard/ParticipantSets/index.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    render: (count: number) => (
+      <Button
+        className={'count-button'}
+        type="text"
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+        }}
+      >
+        <svg
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/index.tsx:80]: There are directories where there are svg s

File: src/components/UserDashboard/ParticipantSets/index.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+const align = 'right' as AlignType;
+
+const columns = [
+  {
+    title: 'Name',
+    dataIndex: 'name',
+    key: 'name',
+    // eslint-disable-next-line react/display-name
+    render: (name: string) => (
+      <div style={{ display: 'inline-flex', alignItems: 'center' }}>
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/index.tsx:56]: css vs style

File: src/components/UserDashboard/ParticipantSets/index.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        <div className={'save-set-table-name'}>{name} </div>
+        <Button size={'small'} type={'text'}>
+          <EditFilled style={{ paddingLeft: 2.75 }} />
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/index.tsx:59]: css vs style

is it possible to have a fraction of an integer for a pixel? 2.75

File: src/components/UserDashboard/ParticipantSets/index.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  userSaveSets: (userId: string) => dispatch(getUserSaveSets(userId)),
+});
+
+const connector = connect(mapState, mapDispatch);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type Props = PropsFromRedux & OwnProps;
+
+const align = 'right' as AlignType;
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/index.tsx:47]: way casting as AlignType?

File: src/components/UserDashboard/index.js [hunk 7]
```diff
@@ -25,11 +25,12 @@ import {
   userDashboardContent,
 } from './UserDashboard.module.css';
 import { Col, Row } from 'antd';
+import ParticipantSets from './ParticipantSets';
 
 const Container = ({ className = '', children }) => (
   // This is to cancel out the negative margin set by react-grid-system
   <div style={{ marginLeft: '15px', marginRight: '15px' }}>
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/index.js:32]: if possible, css vs style (even if it s not your code)

File: src/store/actionCreators/saveSets.ts [hunk 8]
```diff
@@ -66,6 +69,26 @@ export const createSaveSetIfUnique = (
...
+    const payload: UserSaveSets[] = userSets.map(
+      (s: { node: { setId: string; size: number; tag: string } }) => ({
+        setId: s.node.setId,
+        size: s.node.size,
+        tag: s.node.tag,
+      }),
+    );
+    dispatch(displayUserSaveSets(payload));
+  } catch (e) {
+    console.error(e);
```
Comment (evans-g-crsj) [inline src/store/actionCreators/saveSets.ts:87]: console.error vs dispatch failure?

File: src/components/CohortBuilder/ParticipantsTableView/Test/SaveSetModal.test.tsx [hunk 9]
```diff
@@ -6,18 +6,24 @@ import { jestPatchMatchMedia } from '../../../../utils';
...
 import configureStore from 'redux-mock-store';
 import thunk from 'redux-thunk';
-import { SaveSetState } from 'store/saveSetTypes';
+import { SaveSetState, UserSaveSets } from 'store/saveSetTypes';
 
 configure({ adapter: new Adapter() });
 
 const middleware = [thunk];
 const mockStore = configureStore(middleware);
+const userSaveSets = [] as UserSaveSets[];
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/ParticipantsTableView/Test/SaveSetModal.test.tsx:15]: why casting here? can't you write this instead : userSaveSets: UserSaveSets = [] ?

File: src/components/UserDashboard/ParticipantSets/index.tsx (hunk 4 above)
Comment (adipaul1981) [reply src/components/UserDashboard/ParticipantSets/index.tsx:56]: i've tried here. i cant . it is overridden.

File: src/components/UserDashboard/ParticipantSets/index.tsx (hunk 6 above)
Comment (adipaul1981) [reply src/components/UserDashboard/ParticipantSets/index.tsx:47]: thats what ant wants. gives an error if not.

File: src/components/UserDashboard/index.js (hunk 7 above)
Comment (adipaul1981) [reply src/components/UserDashboard/index.js:32]: i think there is task specific to changing the dashboard. i think is a better opportunity to change all this.

File: src/components/UserDashboard/ParticipantSets/index.tsx (hunk 3 above)
Comment (adipaul1981) [reply src/components/UserDashboard/ParticipantSets/index.tsx:80]: well yes. There is already one for this icon. however i had to create a special one for the save set card. I could not change the color from gray to my desired magenta. i could add it to the svg folder. However, it is not a svg that should be used anywhere else.

File: src/components/CohortBuilder/ParticipantsTableView/Test/SaveSetModal.test.tsx (hunk 9 above)
Comment (adipaul1981) [reply src/components/CohortBuilder/ParticipantsTableView/Test/SaveSetModal.test.tsx:15]: nope type is required.

File: src/components/UserDashboard/ParticipantSets/index.tsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    userSaveSets(user.egoId);
+  }, [userSaveSets, user]);
+
+  const data = userSets.sets.map((s) => ({
+    key: s.setId,
+    name: s.tag,
+    count: s.size,
+  }));
+
+  return (
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/ParticipantSets/index.tsx:106]: Personally, I prefer if/else thant multiple ternary operators...but it s a matter of taste.

**Pull request merged**